fabric_plugin/tunnel.py

```diff
@@ -42,22 +39,27 @@ def accept(channel, (src_addr, src_port), (dest_addr, dest_port)):
         try:
             sock.connect((local_host, local_port))
         except Exception as e:
+            try:
+                channel.close()
+            except Exception as e2:
+                close_error = ' (While trying to close channel: {0})'.format(
+                    e2)
+            else:
+                close_error = ''
             raise NonRecoverableError(
-                '[{0}] rtunnel: cannot connect to {1}:{2} ({3})'.format(
+                '[{0}] rtunnel: cannot connect to {1}:{2} ({3}){4}'.format(
                     fabric_api.env.host_string, local_host,
-                    local_port, e.message))
-            channel.close()
-            return
+                    local_port, e, close_error))
 
         th = ThreadHandler('fwd', _forwarder, channel, sock)
         threads.append(th)
 
     transport = connections[fabric_api.env.host_string].get_transport()
-    transport.request_port_forward(
+    remote_port = transport.request_port_forward(
```

Comment:
lol. this is sort of weird. The docstring of `request_port_foward` specifically states that it returns the allocated port on the machine and instead of using it in the tunnel implementation it is simply ignoring it. It's like the guy intentionally decided not to solve a problem here.